catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CreateOperations.java

```diff
@@ -0,0 +1,589 @@
+            throws IngestException, SourceUnavailableException {
+        boolean catalogStoreRequest = opsCrudSupport.isCatalogStoreRequest(createRequest);
+        queryOperations.setFlagsOnRequest(createRequest);
+
+        validateCreateRequest(createRequest);
+
+        if (Requests.isLocal(createRequest)
+                && !sourceOperations.sourceIsAvailable(catalogSupplier.get())) {
+            SourceUnavailableException sourceUnavailableException = new SourceUnavailableException(
+                    "Local provider is not available, cannot perform create operation.");
+            if (INGEST_LOGGER.isWarnEnabled()) {
+                INGEST_LOGGER.warn(
+                        "Error on create operation, local provider not available. {} metacards failed to ingest. {}",
+                        createRequest.getMetacards()
+                                .size(),
+                        buildIngestLog(createRequest),
+                        sourceUnavailableException);
+            }
+            throw sourceUnavailableException;
+        }
+
+        CreateResponse createResponse = null;
+
+        Exception ingestError = null;
+        try {
+            createRequest = injectAttributes(createRequest);
+
+            setDefaultValues(createRequest);
+
+            Map<String, Serializable> unmodifiablePropertiesMap = Collections.unmodifiableMap(
+                    createRequest.getProperties());
+            HashMap<String, Set<String>> requestPolicyMap = new HashMap<>();
+            for (Metacard metacard : createRequest.getMetacards()) {
+                HashMap<String, Set<String>> itemPolicyMap = new HashMap<>();
+                for (PolicyPlugin plugin : frameworkProperties.getPolicyPlugins()) {
+                    PolicyResponse policyResponse = plugin.processPreCreate(metacard,
+                            unmodifiablePropertiesMap);
+                    opsSecuritySupport.buildPolicyMap(itemPolicyMap,
+                            policyResponse.itemPolicy()
+                                    .entrySet());
+                    opsSecuritySupport.buildPolicyMap(requestPolicyMap,
+                            policyResponse.operationPolicy()
+                                    .entrySet());
+                }
+
+                metacard.setAttribute(new AttributeImpl(Metacard.SECURITY, itemPolicyMap));
+            }
+            createRequest.getProperties()
+                    .put(PolicyPlugin.OPERATION_SECURITY, requestPolicyMap);
+
+            for (AccessPlugin plugin : frameworkProperties.getAccessPlugins()) {
+                createRequest = plugin.processPreCreate(createRequest);
+            }
+
+            createRequest.getProperties()
+                    .put(Constants.OPERATION_TRANSACTION_KEY,
+                            new OperationTransactionImpl(OperationTransaction.OperationType.CREATE,
+                                    new ArrayList<>()));
```

Comment:
The transaction previous metacard state list will be immutable.

https://github.com/codice/ddf/blob/6723c7c87888afa1848d046d442345732139931e/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/OperationTransactionImpl.java#L37